crates/astria-sequencer-validation/src/proof.rs

```diff
@@ -70,55 +62,27 @@ pub struct InclusionProof {
     inclusion_proof: CtInclusionProof<Sha256>,
 }
 
-impl Serialize for InclusionProof {
-    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let mut s = serializer.serialize_struct("InclusionProof", 4)?;
-        s.serialize_field("value", &hex::encode(&self.value))?;
-        s.serialize_field("idx", &self.idx)?;
-        s.serialize_field("num_leaves", &self.num_leaves)?;
-        s.serialize_field(
-            "inclusion_proof",
-            &hex::encode(self.inclusion_proof.as_bytes()),
-        )?;
-        s.end()
+impl PartialEq for InclusionProof {
+    fn eq(&self, other: &Self) -> bool {
+        self.idx == other.idx
+            && self.num_leaves == other.num_leaves
+            && self.inclusion_proof.as_bytes() == other.inclusion_proof.as_bytes()
     }
 }
 
-impl<'de> Deserialize<'de> for InclusionProof {
-    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
-        #[derive(Deserialize)]
-        struct InclusionProofHelper {
-            value: String,
-            idx: usize,
-            num_leaves: usize,
-            inclusion_proof: String,
-        }
-        let helper = InclusionProofHelper::deserialize(deserializer)?;
-        let value = hex::decode(helper.value).map_err(serde::de::Error::custom)?;
-        let inclusion_proof = CtInclusionProof::try_from_bytes(
-            hex::decode(helper.inclusion_proof).map_err(serde::de::Error::custom)?,
-        )
-        .map_err(serde::de::Error::custom)?;
-        Ok(InclusionProof {
-            value,
-            idx: helper.idx,
-            num_leaves: helper.num_leaves,
-            inclusion_proof,
-        })
-    }
-}
+impl Eq for InclusionProof {}
 
 impl InclusionProof {
     /// Verify that the proof is valid for the given root hash.
     ///
     /// # Errors
     ///
     /// - if the proof is invalid
-    pub fn verify(&self, root_hash: Hash) -> eyre::Result<()> {
+    pub fn verify(&self, value: &[u8], root_hash: Hash) -> eyre::Result<()> {
```

Comment:
Does this still match the doc comment? What is value?

This method name is confusing to me. Is there proof verified? Is the data verified? Can you add a doc test that demonstrates how this should be used?

Reply:
updated the comment slightly - it's verifying a merkle proof of inclusion, so proving that the `value` is a leaf inside the merkle tree where the root is `root_hash`. not sure how much info I should add, since I thought the type was somewhat explanatory haha

can add a doc comment though